samza-core/src/main/java/org/apache/samza/processor/StreamProcessorLifeCycleAware.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.processor;
+
+import org.apache.samza.annotation.InterfaceStability;
+
+
+/**
+ * This class listens to the life cycle events in a {@link StreamProcessor},
+ * and triggers the corresponding callbacks.
+ */
+
+@InterfaceStability.Evolving
+public interface StreamProcessorLifeCycleAware {
```

Comment:
Is this a user-facing interface for lifecycle ? Or is it simply internal? 

Even though this class listens to the lifecycle events in StreamProcessor, the interaction seems to leak into SamzaContainerController class. This breaks the logical abstractions where StreamProcessor is responsible for the lifecycle of container and coordinator.